cmd/test-image/server/server.go

```diff
@@ -0,0 +1,104 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+)
+
+var basePaths = map[string]HTTPHandlerFunc{
+	"/":      notImplemented,
+	"/ready": ready,
+}
+
+type HTTPHandlerFunc func(s *TestServer, w http.ResponseWriter, r *http.Request)
+
+type httpHandler struct {
+	server *TestServer
+	F      HTTPHandlerFunc
+}
+
+func (h httpHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	h.F(h.server, w, r)
+}
+
+type TestServer struct {
+	srv               *http.Server
+	port              int
+	mux               *http.ServeMux
+	registeredHanders map[string]bool
+	sync.Mutex
+}
+
+// New creates a new diagnostic server
+func NewTestServer() *TestServer {
+	return &TestServer{
+		registeredHanders: make(map[string]bool),
+	}
+}
+
+// Init initialize the mux for the http handling and register the base hooks
+func (s *TestServer) Init() {
+	s.mux = http.NewServeMux()
+
+	// Register local handlers
+	s.RegisterHandler(basePaths)
+}
+
+// RegisterHandler allows to register new handlers to the mux and to a specific path
+func (s *TestServer) RegisterHandler(hdlrs map[string]HTTPHandlerFunc) {
+	s.Lock()
+	defer s.Unlock()
+	for path, fun := range hdlrs {
+		if _, ok := s.registeredHanders[path]; ok {
+			continue
+		}
+		s.mux.Handle(path, httpHandler{s, fun})
+		s.registeredHanders[path] = true
+	}
+}
+
```

Comment:
shouldnt golint fail if we dont have comment for exported function?

Reply:
sure this was a very first draft